### DISCUSSION TITLE
How does one change the contents of the Properties tool window?

### DISCUSSION THREAD

**Discussion by akorchev** [VsixCommunity/Community.VisualStudio.Toolkit#496] [Q&A]
Hi all!

I have a VS extension and need to create a custom property grid implementation. Customizing the default one doesn't look ideal and some things don't seem possible (for example filtering properties with search as you type).

I can do it via custom `ToolWindowPane`. This would require my users to show it from the View -> Other Windows menu. I would much prefer to replace the contents of the Properties window instead. This seems possible as the WPF component properties are  different.  I've checked the actual components displayed: 

`Microsoft.VisualStudio.Platform.WindowManagement.PropertyBrowserSwitcher` (default old-school Property Grid) 
<img width="428" alt="image" src="https://github.com/VsixCommunity/Community.VisualStudio.Toolkit/assets/454726/26639806-2de2-4c6f-834c-7c04e3515719">

and `Microsoft.VisualStudio.DesignTools.Designer.UI.PropertyInspector.SceneNodePropertyInspectorPane` (the fancy WPF grid)
<img width="685" alt="image" src="https://github.com/VsixCommunity/Community.VisualStudio.Toolkit/assets/454726/eac6ef27-cdae-453c-8ed8-7e0ec2b1a1c6">

Does anybody know how this is done?

**Comment by madskristensen** [comment]
@AArnott do you know where to start on this?

**Reply by AArnott** [reply]
No, I'm afraid not.

**Comment by vincedan** [comment]
@akorchev , I have the same question. do you have the solution right now?
